libs/concurrency/include/hpx/concurrency/cache_line_data.hpp

```diff
@@ -41,55 +41,41 @@ namespace hpx {
             }
         }    // namespace detail
 
+        // NOTE: We do not use alignas here because asking for overaligned
+        // memory is significantly more expensive than asking for unaligned
+        // memory. Padding the struct is cheaper and enough for internal
+        // purposes.
+
+        // NOTE: The implementations below are currently identical because of
+        // the above issue. Both names are kept for compatibility.
+
         ///////////////////////////////////////////////////////////////////////////
         // special struct to ensure cache line alignment of a data type
-#if defined(HPX_HAVE_CXX11_ALIGNAS) && defined(HPX_HAVE_CXX17_ALIGNED_NEW) &&  \
-    !defined(__NVCC__)
-        template <typename Data>
-        struct alignas(threads::get_cache_line_size()) cache_aligned_data
-        {
-            Data data_;
-        };
-#else
         template <typename Data>
         struct cache_aligned_data
         {
+            cache_aligned_data()
+              : data_{}
+            {
+            }
+            cache_aligned_data(Data data)
```

Comment:
You might want to consider adding two constructors instead, one taking a (const) lvalue ref and one taking an rvalue reference.

Reply:
Added the constructors.